app/src/composables/use-relation-multiple.ts

```diff
@@ -178,12 +178,15 @@ export function useRelationMultiple(
 
 		items.push(...fullSelectedOnPage, ...newItems);
 
-		const sortField = relation.value.sortField;
+		const sortField = (previewQuery.value.sort ?? toArray(relation.value.sortField))[0];
 
 		if ((previewQuery.value.limit > 0 && totalItemCount.value > previewQuery.value.limit) || !sortField) return items;
 
 		return items.sort((a, b) => {
-			return a[sortField] - b[sortField];
+			const field = sortField.replace('-', '');
+
+			if (sortField.startsWith('-')) return get(b, field) - get(a, field);
+			else return get(a, field) - get(b, field);
```

Comment:
Although `replace('-', '')` does technically works, it might be better to follow our existing convention of `substring(1)` 👍 such as:

- https://github.com/directus/directus/blob/9526b4e5b222acc8f58aa3ad61fb15393dca74e8/api/src/database/run-ast.ts#L490-L493

- https://github.com/directus/directus/blob/9526b4e5b222acc8f58aa3ad61fb15393dca74e8/api/src/utils/apply-query.ts#L262-L264

perhaps especially for edge cases where a setup may (somehow) have fields with hyphens (eg. `my-new-field`), even if such cases aren't truly officially supported per se.

Reply:
```suggestion
			const field = sortField.substring(1);

			if (sortField.startsWith('-')) return get(b, field) - get(a, field);
			else return get(a, field) - get(b, field);
```

Reply:
Updated it